modules/core/04-channel/keeper/upgrade.go

```diff
@@ -257,6 +257,33 @@ func (k Keeper) startFlushUpgradeHandshake(
 	return nil
 }
 
+// WriteUpgradeAckChannel writes a channel which has successfully passed the UpgradeAck handshake step as well as
+// setting the upgrade for that channel.
+// An event is emitted for the handshake step.
+func (k Keeper) WriteUpgradeAckChannel(
+	ctx sdk.Context,
+	portID, channelID string,
+	proposedUpgrade types.Upgrade,
+	flushStatus types.FlushStatus,
```

Comment:
I think its better to determine flushStatus inside this function rather than passing it in as argument

Reply:
You can do this by creating an iterator over packet commitments and seeing if it is empty

Reply:
okay, that makes sense!

Reply:
We can add the iterator logic to determine that in a follow up! :)

Reply:
ah, i just used the `GetAllPacketCommitmentsAtChannel` method from the 04-channel keeper, is there a reason why this should not suffice?

Reply:
that should work since its doing the work for you. Please add tests to double check. Nice work keeping it DRY